pkg/api/customization/namespace/namespace.go

```diff
@@ -44,3 +52,46 @@ func ProjectMap(apiContext *types.APIContext, refresh bool) (map[string]string,
 
 	return data, nil
 }
+
+type ActionWrapper struct {
+	ClusterManager *clustermanager.Manager
+}
+
+func (w ActionWrapper) ActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	actionInput, err := parse.ReadBody(apiContext.Request)
+	if err != nil {
+		return err
+	}
+	switch actionName {
+	case "move":
+		clusterID, _ := ref.Parse(convert.ToString(actionInput["projectId"]))
+		userContext, err := w.ClusterManager.UserContext(clusterID)
+		if err != nil {
+			return err
+		}
+		nsClient := userContext.Core.Namespaces("")
+		ns, err := nsClient.Get(apiContext.ID, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		ns.Annotations[projectIDFieldLabel] = convert.ToString(actionInput["projectId"])
```

Comment:
Add a safety check, if the annotation exists we shouldn't be moving the namespace